### PR TITLE
[Merged by Bors] - fix: improve line-wrapping of (D)Finsupp reprs

### PR DESCRIPTION
**Pull request by eric-wieser**
These were previously wrapping very poorly.

Also update tests to use `#guard_msgs` more effectively, and fix a typo in the delaborators for DFinsupp.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: test/dfinsupp_notation.lean [hunk 1]
```diff
@@ -7,20 +7,23 @@ example : (fun₀ | 1 | 2 | 3 => 3 | 3 => 4 : Π₀ i, Fin (i + 10)) 1 = 3 := by
 example : (fun₀ | 1 | 2 | 3 => 3 | 3 => 4 : Π₀ i, Fin (i + 10)) 2 = 3 := by simp
 example : (fun₀ | 1 | 2 | 3 => 3 | 3 => 4 : Π₀ i, Fin (i + 10)) 3 = 4 := by simp
 
-/--
-info:
--/
+/-- info: (DFinsupp.single 1 3).update 2 3 : Π₀ (i : ℕ), Fin (i + 10) -/
```
Comment (Vierkantor) [inline test/dfinsupp_notation.lean:10]: Why did this get broken? Before it would say `fun₀ | 1 => 3 | 2 => 3`.

File: test/finsupp_notation.lean [hunk 2]
```diff
@@ -12,14 +12,49 @@ example : (fun₀ | 1 | 2 | 3 => 3 | 3 => 4) 2 = 3 := by
...
+/-- info: fun₀ | 1 => 3 | 2 => 3 -/
+#guard_msgs in
+#eval (Finsupp.mk {1, 2} (fun | 1 | 2 => 3 | _ => 0) (fun x => by aesop))
+
 /--
 info:
+fun₀
+| ["there are five words here", "and five more words here"] => 5
+| ["there are seven words but only here"] => 7
+| ["just two"] => 2
```
Comment (Vierkantor) [inline test/finsupp_notation.lean:26]: Indentation doesn't match between the `#eval` and the `#check` command, can you fix that?

Comment (eric-wieser) [reply test/finsupp_notation.lean:26]: Which version do you prefer?

Comment (Vierkantor) [reply test/finsupp_notation.lean:26]: I suppose the two-space-indented version is a bit clearer, but either works for me.

File: test/dfinsupp_notation.lean (hunk 1 above)
Comment (eric-wieser) [reply test/dfinsupp_notation.lean:10]: Ah, I accidentally replaced `eval` with `check`. I guess `check` was already broken.

Comment (eric-wieser) [reply test/dfinsupp_notation.lean]: Fixed.

File: test/finsupp_notation.lean (hunk 2 above)
Comment (eric-wieser) [reply test/finsupp_notation.lean]: Done.

**Review by Vierkantor — APPROVED** [review]
Nice work!

bors r+

**Pull request closed without merging**